Core/Utilities/MySQL.py

```diff
@@ -331,131 +330,125 @@ def __pop( self, thid ):
-      for thid in list( self.__assigned ):
+      for thid in list(self.__assigned):
         if not thid.isAlive():
-          self.__pop( thid )
+          self.__pop(thid)
         try:
-          data = self.__assigned[ thid ]
+          data = self.__assigned[thid]
         except KeyError:
           continue
         if now - data[2] > self.__graceTime:
-          self.__pop( thid )
+          self.__pop(thid)
 
-    def transactionStart( self, dbName ):
-      result = self.get( dbName )
-      if not result[ 'OK' ]:
+    def transactionStart(self, dbName):
+      result = self.get(dbName)
+      if not result['OK']:
         return result
-      conn = result[ 'Value' ]
+      conn = result['Value']
       try:
-        return S_OK( self.__execute( conn, "START TRANSACTION WITH CONSISTENT SNAPSHOT" ) )
-      except MySQLdb.MySQLError, excp:
-        return S_ERROR( DErrno.EMYSQL, "Could not begin transaction: %s" % excp )
+        return S_OK(self.__execute(conn, "START TRANSACTION WITH CONSISTENT SNAPSHOT"))
+      except MySQLdb.MySQLError as excp:
+        return S_ERROR(DErrno.EMYSQL, "Could not begin transaction: %s" % excp)
 
-    def transactionCommit( self, dbName ):
-      result = self.get( dbName )
-      if not result[ 'OK' ]:
+    def transactionCommit(self, dbName):
+      result = self.get(dbName)
+      if not result['OK']:
         return result
-      conn = result[ 'Value' ]
+      conn = result['Value']
       try:
-        result = self.__execute( conn, "COMMIT" )
-        return S_OK( result )
-      except MySQLdb.MySQLError, excp:
-        return S_ERROR( DErrno.EMYSQL, "Could not commit transaction: %s" % excp )
-
-    def transactionRollback( self, dbName ):
-      result = self.get( dbName )
-      if not result[ 'OK' ]:
+        result = self.__execute(conn, "COMMIT")
+        return S_OK(result)
+      except MySQLdb.MySQLError as excp:
+        return S_ERROR(DErrno.EMYSQL, "Could not commit transaction: %s" % excp)
+
+    def transactionRollback(self, dbName):
+      result = self.get(dbName)
+      if not result['OK']:
         return result
-      conn = result[ 'Value' ]
+      conn = result['Value']
       try:
-        result = self.__execute( conn, "ROLLBACK" )
-        return S_OK( result )
-      except MySQLdb.MySQLError, excp:
-        return S_ERROR( DErrno.EMYSQL, "Could not rollback transaction: %s" % excp )
+        result = self.__execute(conn, "ROLLBACK")
+        return S_OK(result)
+      except MySQLdb.MySQLError as excp:
+        return S_ERROR(DErrno.EMYSQL, "Could not rollback transaction: %s" % excp)
 
   __connectionPools = {}
 
-  def __init__( self, hostName = 'localhost', userName = 'dirac', passwd = 'dirac', dbName = '', port = 3306, debug = False ):
+  def __init__(self, hostName='localhost', userName='dirac', passwd='dirac', dbName='', port=3306, debug=False):
     """
     set MySQL connection parameters and try to connect
+
+    :param debug: unused
```

Comment:
should it be removed then altogether ?

Reply:
I would have wanted to,but there is no way to "deprecate it" with the decorator as it is now. I will try to find a way fo a further patch release, and we remove it in v7. OK for you ?

Reply:
OK